### DISCUSSION TITLE
WebGLPointsLayer features selections

### DISCUSSION THREAD

**Discussion by hmr23zfd** [openlayers/openlayers#15126] [Q&A] [answered]
Dear all,
I am using WebGLPointsLayer Layer to render some data. I have a lot of overlapping features on the map. I want to select all overlapping features when the user clicks on them.
is it possible?
Thanks in advance.

**Comment by jahow** [comment, marked as answer]
Hi!

Currently hit detection works by looking at the first feature found under the cursor. Other objects hidden behind it will not be picked up. Depending on how many objects you have and if they are spatially indexed, you could use the `getFeaturesAtCoordinate` method available on vector sources, but that will most likely take more time.

**Reply by hmr23zfd** [reply]
Thanks for your quick reply.

**Reply by hmr23zfd** [reply]
Hi, 
The getFeaturesAtCoordinate method of vector source always return an empty array.
